lib/Logging.js

```diff
@@ -0,0 +1,149 @@
+        };
+        const formatterFn = function(opts) {
+            return opts.timestamp() + ' ' +
+            opts.level.toUpperCase() + ':' +
+            (opts.meta && opts.meta.loggerName ? opts.meta.loggerName : "") + ' ' +
+            (undefined !== opts.message ? opts.message : '');
+        };
+        // if (this.transports.length > 0) {
+        //     updatingLoggers = true;
+        //     this.transports.forEach((transport) => {
+        //         this.loggers.forEach((wrapper, name) => {
+        //             wrapper.logger.remove(transport);
+        //         });
+        //     });
+        // }
+        this.transports = [];
+        if (config.console != undefined && config.console != "off") {
+            this.transports.push(new (winston.transports.Console)({
+                json: false,
+                name: "console",
+                timestamp: timestampFn,
+                formatter: formatterFn,
+                level: config.console
+            }));
+        }
+
+        if(config.files == undefined) {
+            return;
+        }
+
+        let i = 0;
+        for (let file of config.files) {
+            const filename = Object.keys(file)[0];
+            const level = file[filename];
+            i++;
+            this.transports.push(new (winston.transports.DailyRotateFile)({
+                filename,
+                datePattern: "YYYY-MM-DD",
+                name: `logfile` + i,
+                formatter: formatterFn,
+                level,
+                timestamp: timestampFn,
+                maxFiles: config.maxFiles > 0 ? config.maxFiles : undefined
+            }));
+        }
+
+        this.loggers.forEach((wrapper, name) => {
+            wrapper.SwapIn(this.createLogger(name));
+        });
+    }
+
+    Get(name) {
+        if(!this.loggers.has(name)) {
+            const wrapper = new LogWrapper()
+            this.loggers.set(name, wrapper);
+            if (this.transports != null) {
+                wrapper.SwapIn(this.createLogger(name));
+            }
+        }
+        return this.loggers.get(name);
+    }
+
+    createLogger(name) {
+        const logger =  new (winston.Logger)({
+            transports: this.transports,
+            // winston doesn't support getting the logger category from the
+            // formatting function, which is a shame. Instead, write a rewriter
+            // which sets the 'meta' info for the logged message with the loggerName
+            rewriters: [
+                function(level, msg, meta) {
+                    if (!meta) { meta = {}; }
+                    meta.loggerName = name;
+                    return meta;
+                }
+            ]
+        });
+        return logger;
+    }
+}
+
+class LogWrapper {
+    constructor() {
+        this.logger = null;
+        this.messages = [];
+    }
+
+    SwapIn(logger) {
+        this.logger = logger;
+    }
+
+    debug(...message) {this._log(message, 'debug')};
```

Comment:
`message` is a bit of a misnomer; there could be multiple arguments (hence the spread operator `...`)